src/main/java/org/ibp/api/rest/labelprinting/LotLabelPrinting.java

```diff
@@ -0,0 +1,376 @@
+import org.ibp.api.rest.labelprinting.domain.OriginResourceMetadata;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.context.support.ResourceBundleMessageSource;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.ObjectError;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Component
+@Transactional
+public class LotLabelPrinting extends LabelPrintingStrategy {
+
+	@Autowired
+	private ResourceBundleMessageSource messageSource;
+
+	@Autowired
+	private GermplasmDataManager germplasmDataManager;
+
+	@Autowired
+	private SearchRequestService searchRequestService;
+
+	@Autowired
+	private LotService lotService;
+
+	public static List<FileType> SUPPORTED_FILE_TYPES = Arrays.asList(FileType.CSV, FileType.PDF, FileType.XLS);
+
+
+	// Lot IDs
+	// (Keeping it TERM_ID agnostic for now)
+	// TODO generic lot variable columns query
+	private enum LOT_FIELD {
+		LOT_ID(22, "Lot ID"), // Added later
+		LOT_UID(1, "Lot UID"),
+		STOCK_ID(2, "Stock id"),
+		STATUS(3, "Status"),
+		STORAGE_LOCATION(4, "Storage location"),
+		UNITS(5, "Units"),
+		ACTUAL_BALANCE(6, "Actual balance"),
+		AVAILABLE(7, "Available"),
+		RESERVED(8, "Reserved"),
+		TOTAL_WITHDRAWALS(9, "Total withdrawals"),
+		PENDING_DEPOSITS(10, "Pending deposits"),
+		NOTES(11, "Notes"),
+		USERNAME(12, "Username"),
+		CREATION_DATE(13, "Creation date"),
+		LAST_DEPOSIT_DATE(14, "Last deposit date"),
+		LAST_WITHDRAWAL_DATE(15, "Last withdrawal date");
+
+		private static Map<Integer, LOT_FIELD> byId =
+			Arrays.stream(LOT_FIELD.values()).collect(Collectors.toMap(LOT_FIELD::getId, Function.identity()));
+
+		private final int id;
+		private final String name;
+
+		LOT_FIELD(final int id, final String name) {
+			this.id = id;
+			this.name = name;
+		}
+
+		public int getId() {
+			return this.id;
+		}
+
+		public String getName() {
+			return this.name;
+		}
+
+		public static LOT_FIELD getById(final int id) {
+			return byId.get(id);
+		}
+	}
+
+
+	// Lot fields
+	private static LabelType LOT_FIXED_LABEL_TYPES = new LabelType("Lot", "Lot")
+		.withFields(Arrays.stream(LOT_FIELD.values())
+			.map(field -> new Field(field.getId(), field.getName()))
+			.collect(Collectors.toList()));
+
+
+	// Germplasm ids
+	// (Keeping it TERM_ID agnostic for now)
+	private enum GERMPLASM_FIELD {
+		GID(16, "GID"),
+		GROUP_ID(17, "Group ID"),
+		DESIGNATION(18, "Designation"),
+		BREEDING_METHOD(19, "Breeding method"),
+		LOCATION(20, "Location"),
+		CROSS(21, "Cross")
+		// LOT_ID(22) in lot group
```

Comment:
do we need to keep this comment?

Reply:
@clarysabel it does't harm and it makes you aware that you cannot use the id 22 at that point and you need to skip it